mpc-core/src/utils.rs

```diff
@@ -65,6 +66,18 @@ pub fn parse_ristretto_key(b: Vec<u8>) -> Result<RistrettoPoint, std::io::Error>
     }
 }
 
+pub fn sha256(data: &[u8]) -> [u8; 32] {
+    let mut hasher = Sha256::new();
+    hasher.update(data);
+    hasher.finalize().into()
+}
+
+/// XORs 2 slices of bytes of equal length
+pub fn xor(a: &[u8], b: &[u8]) -> Vec<u8> {
```

Comment:
This always results in a heap allocation (Vec) even if the iterables are on the stack. We could instead pass in a mutable reference:

```rust
fn xor(a: &[u8], b: &[u8], out: &mut [u8]) {
    for ((a, b), out) in a.iter().zip(b.iter()).zip(out.iter_mut()) {
        *out = a ^ b;
    }
}
```

Or we could alternatively avoid any copies/allocation

```rust
fn xor_replace(a: &mut [u8], b: &[u8]) {
    for (a, b) in a.iter_mut().zip(b.iter()) {
        *a ^= b;
    }
}
```